mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,806 @@
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y: ArrayLike of shape (n_samples,)
+            Training labels.
+
+        groups: Optional[ArrayLike] of shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+            By default ``None``.
+
+        z: Optional[ArrayLike] of shape (n_calib_samples, n_exog_features)
+            Exogenous variables
+
+            By default ``None``
+
+        alpha: Optional[float]
+            Between ``0.0`` and ``1.0``, represents the risk level of the
+            confidence interval.
+            Lower ``alpha`` produce larger (more conservative) prediction
+            intervals.
+            ``alpha`` is the complement of the target coverage level.
+
+            If ``None``, the calibration will be done using the ``alpha``value
+            set in the initialisation. Else, the new value will overwrite the
+            old one.
+
+            By default ``None``
+
+        Returns
+        -------
+        MapieCCPRegressor
+            self
+
+        """
+        self._check_fit_parameters()
+        self._check_calibrate_parameters()
+
+        self.estimator = cast(RegressorMixin, self.estimator)
+        self.cv = cast(Union[str, BaseCrossValidator], self.cv)
+        self.conformity_score_ = cast(ConformityScore, self.conformity_score_)
+
+        if self.cv != 'prefit':
+            try:
+                if isinstance(self.estimator, Pipeline):
+                    check_is_fitted(self.estimator[-1])
+                else:
+                    check_is_fitted(self.estimator)
+            except NotFittedError as exc:
+                raise NotFittedError("As you are using an estimator which is "
+                                     "not fitted yet, you need to call the "
+                                     "fit method before calibrate.") from exc
+
+            calib_index = list(
+                cast(BaseCrossValidator, self.cv).split(X, y, groups)
+            )[0][1]
+            X_calib = cast(NDArray, _safe_indexing(X, calib_index))
+            y_calib = cast(NDArray, _safe_indexing(y, calib_index))
+            if z is not None:
+                z_calib = cast(NDArray, _safe_indexing(z, calib_index))
+            else:
+                z_calib = None
+        else:
+            X_calib = cast(NDArray, X)
+            y_calib = cast(NDArray, y)
+            if z is not None:
+                z_calib = cast(NDArray, z)
+            else:
+                z_calib = None
+
+        if alpha is not None and self.alpha != alpha:
+            self.alpha = self._check_alpha(alpha)
+            warnings.warn(f"WARNING: The old value of alpha "
+                          f"({self.alpha}) has been overwritten "
+                          f"by the new one ({alpha}).")
+
+        self.phi._check_need_calib(X_calib)
```

Comment:
In my previous comment, this must be move / remove by replacing X_calib by X_all.